### PR TITLE
DUPLO-12737 & DUPLO-14306: k8s job fixes

### PR DESCRIPTION
**Pull request by ddiaz-duplo**
# Overview

- [DUPLO-12737](https://app.clickup.com/t/8655600/DUPLO-12737)
  - Fix job Update
    - We were previously not passing tenantId, the update would fail, and we recreated the resource.
  - Fix labels
    - Exclude duplo generated labels. Apply diff to anything else.
- [DUPLO-14306](https://app.clickup.com/t/8655600/DUPLO-14306)
  - Fixed reading data-source
  - added example
- [DUPLO-14305](https://app.clickup.com/t/8655600/DUPLO-14305)
  - add data source examples

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Testing
testing shows update of label
image

data source fix
```terraform
data "duplocloud_k8s_job" "job" {
  tenant_id = "604831b1-9972-4a6a-acef-06d7b55d9b8e"
  metadata {
    name = "datajob"
  }
  spec {
    template {
    }
  }
}

output "metadata" {
  value = data.duplocloud_k8s_job.job.metadata[0].namespace
}
```
image

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-12737 TF: K8s job/Cron Job: don't force replacement of k8s job on labels ,annotations and container remove  change](https://app.clickup.com/t/8655600/DUPLO-12737)

File: examples/data-sources/duplocloud_k8s_job/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+data "duplocloud_k8s_job" "job" {
+  tenant_id = "604831b1-9972-4a6a-acef-06d7b55d9b8e"
+  metadata {
+    name = "datajob"
+  }
+  spec {
```
Comment (joek-duplo) [inline examples/data-sources/duplocloud_k8s_job/data-source.tf:6]: The spec in a data source must be readonly and thus not specified by the user

**Review by joek-duplo — CHANGES_REQUESTED** [review]
Minor changes

File: examples/data-sources/duplocloud_k8s_job/data-source.tf (hunk 1 above)
Comment (ddiaz-duplo) [reply examples/data-sources/duplocloud_k8s_job/data-source.tf]: fixed.

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**